File: src/backend/app/schemas/billing.py
"""Billing schemas — subscription responses."""

import uuid
from datetime import datetime

from pydantic import BaseModel


class SubscriptionResponse(BaseModel):
    id: uuid.UUID
    tier: str
    status: str
    project_limit: int | None
    current_period_end: datetime | None = None
    trial_ends_at: datetime | None = None

    model_config = {"from_attributes": True}

File: src/frontend/lib/billing.ts
import { apiFetch } from "@/lib/api";

export type Tier = "free" | "starter" | "pro";

export interface Subscription {
  id: string;
  tier: Tier;
  status: string;
  project_limit: number | null;
  current_period_end: string;
  trial_ends_at: string | null;
}

export interface Usage {
  project_count: number;
  user_count: number;
  storage_bytes: number;
}

export interface CheckoutResponse {
  checkout_url: string;
}

export function getSubscription(): Promise<Subscription> {
  return apiFetch<Subscription>("/billing/subscription");
}

export function getUsage(): Promise<Usage> {
  return apiFetch<Usage>("/billing/usage");
}

export function createCheckout(tier: Exclude<Tier, "free">): Promise<CheckoutResponse> {
  return apiFetch<CheckoutResponse>("/billing/checkout", {
    method: "POST",
    body: JSON.stringify({ tier }),
  });
}
